dash-pipeline/bmv2/dash_pipeline.p4

```diff
@@ -231,6 +235,98 @@ control dash_ingress(
         const default_action = deny;
     }
 
+    action set_meter_policy_attrs(@Sai[type="sai_ip_addr_family_t", isresourcetype="true"] bit<32> ip_addr_family) {
+        if (ip_addr_family == 0) /* SAI_IP_ADDR_FAMILY_IPV4 */ {
+            if (meta.is_overlay_ip_v6 == 1) {
+                meta.dropped = true;
+            }
+        } else {
+            if (meta.is_overlay_ip_v6 == 0) {
+                meta.dropped = true;
+            }
+        }
+    }
+
+    @name("meter_policy|dash_meter")
+    @Sai[isobject="true"]
+    table meter_policy {
+        key = {
+            meta.meter_policy_id : exact @name("meta.meter_policy_id:meter_policy_id");
+        }
+        actions = {
+            set_meter_policy_attrs();
+        }
+    }
+
+    action meter_rule_action(bit<16> meter_class) {
+        meta.policy_meter_class = meter_class;
+    }
+
+    @name("meter_rule|dash_meter")
+    @Sai[isobject="true"]
+    table meter_rule {
+        key = {
+            meta.meter_policy_id: exact @name("meta.meter_policy_id:meter_policy_id") @Sai[type="sai_object_id_t", isresourcetype="true", objects="METER_POLICY"];
+            hdr.ipv4.dst_addr : ternary @name("hdr.ipv4.dst_addr:dip");
+        }
+
+     actions = {
+            meter_rule_action;
+            @defaultonly deny;
+        }
+        const default_action = deny;
+    }
+
+    action meter_bucket_action(
+            @Sai[type="sai_uint32_t", isreadonly="true"] bit<32> outbound_bytes_counter,
+            @Sai[type="sai_uint32_t", isreadonly="true"] bit<32> inbound_bytes_counter) {
+        // read only counters for SAI api generation
+    }
+
+    @name("meter_bucket|dash_meter")
+    @Sai[isobject="true"]
+    table meter_bucket {
+        key = {
+            meta.eni_id: exact @name("meta.eni_id:eni_id");
+            meta.meter_class: exact @name("meta.meter_class:meter_class");
+        }
+        actions = {
+            meter_bucket_action;
+        }
+    }
+
+#ifdef TARGET_BMV2_V1MODEL
+    direct_counter(CounterType.bytes) meter_bucket_outbound_counter;
+#endif
+    @name("meter_bucket_outbound|dash_meter")
+    @Sai[ignoretable="true"]
+    table meter_bucket_outbound {
+        key = {
+            meta.meter_class: exact @name("meta.meter_class:meter_class");
+            meta.eni_id: exact @name("meta.eni_id:eni_id");
+        }
+        actions = { NoAction; }
+#ifdef TARGET_BMV2_V1MODEL
+        counters = meter_bucket_outbound_counter;
+#endif
+    }
+
+#ifdef TARGET_BMV2_V1MODEL
+    direct_counter(CounterType.bytes) meter_bucket_inbound_counter;
```

Comment:
For the inbound case, does the meter_class need to be added to the inbound_routing table actions?

Reply:
yes. will address inbound meter in a separate PR